src/ducks/send.ts

```diff
@@ -0,0 +1,100 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import StellarSdk, { Memo } from "stellar-sdk";
+import BigNumber from "bignumber.js";
+import { ActionStatus } from "./account";
+
+// TODO - network constant config
+const server = new StellarSdk.Server("https://horizon-testnet.stellar.org");
+
+export const sendTransaction = createAsyncThunk<
+  any,
+  {
+    secret: string;
+    toAccountId: string;
+    amount: BigNumber;
+    fee: number;
+    memo: Memo;
+  },
+  { rejectValue: RejectMessage }
+>(
+  "sendTransaction",
+  async ({ secret, toAccountId, amount, fee, memo }, { rejectWithValue }) => {
+    let result;
+
+    try {
+      const keypair = StellarSdk.Keypair.fromSecret(secret);
+      const sequence = (await server.loadAccount(keypair.publicKey())).sequence;
+      const source = await new StellarSdk.Account(
+        keypair.publicKey(),
+        sequence,
+      );
+
+      let transaction = new StellarSdk.TransactionBuilder(source, {
+        fee,
+        networkPassphrase: StellarSdk.Networks.TESTNET,
+      }).addOperation(
+        StellarSdk.Operation.payment({
+          destination: toAccountId,
+          asset: StellarSdk.Asset.native(),
+          amount: amount.toString(),
+        }),
+      );
+
+      if (memo.type !== StellarSdk.MemoNone) {
+        transaction = transaction.addMemo(memo);
+      }
+
+      transaction = transaction.setTimeout(StellarSdk.TimeoutInfinite).build();
```

Comment:
@quietbits do we need to use "fetchTimebounds" from the network for submitting transactions here? I noticed mobilewallet is

Reply:
Yes, I think we do. It will give us a more accurate time of user's transaction. https://stellar.github.io/js-stellar-sdk/Server.html#fetchTimebounds